relay-event-normalization/src/sdk_version.rs

```diff
@@ -0,0 +1,148 @@
+use std::num::ParseIntError;
+
+const DEFAULT_VERSION_IF_EMPTY: usize = 0;
+
+/// Represents an SDK Version using the semvar versioning, meaning MAJOR.MINOR.PATCH.
+/// An optional release type can be specified, then it becomes
+/// MAJOR.MINOR.PATCH-(alpha|beta).RELEASE_VERSION
+#[derive(Debug, PartialOrd, Eq, PartialEq, Ord)]
+pub struct SdkVersion {
+    major: usize,
+    minor: usize,
+    patch: usize,
+    release_type: ReleaseType,
+}
+
+/// Represents the release type which might be present in a version string,
+/// for example: 9.0.0-alpha.0.
+/// Release types are also comparable to each other, using the rules:
+/// alpha < beta < release.
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd)]
+enum ReleaseType {
+    Alpha(usize),
+    Beta(usize),
+    Release,
+}
+
+impl SdkVersion {
+    pub const fn new(major: usize, minor: usize, patch: usize) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+            release_type: ReleaseType::Release,
+        }
+    }
+
+    /// Attempts to parse a version string and returning a [`ParseIntError`] if it contains any
+    /// non-numerical characters apart from dots.
+    /// If a version part is not provided, 0 will be assumed.
+    /// For example:
+    /// 1.2 -> 1.2.0
+    /// 1   -> 1.0.0
+    pub fn try_parse(input: &str) -> Result<Self, ParseIntError> {
```

Comment:
Instead of bubbling the `ParseIntError` making a new `SdkVersionParseError` with or without `thiserror` is pretty quick and more informational.